### PR TITLE
Add log file info for all kind of segment binlogs

### PR DESCRIPTION
**Pull request by congqixia**
See also #13262 

/kind enhancement

Signed-off-by: Congqi Xia <congqi.xia@zilliz.com>

### CODE REVIEW

File: internal/proto/data_coord.proto [hunk 1]
```diff
@@ -231,7 +231,7 @@ message SaveBinlogPathsRequest {
   repeated SegmentStartPosition start_positions = 6;
   bool flushed = 7;
   repeated FieldBinlog field2StatslogPaths = 8;
-  repeated DeltaLogInfo deltalogs = 9;
+  repeated Binlog deltalogs = 9;
```
Comment (xiaofan-luan) [inline internal/proto/data_coord.proto:234]: Just wondering, is it better to also FieldBinlog for delta logs and make Binlog a private concept inside FieldBinLog?

Comment (xiaofan-luan) [reply internal/proto/data_coord.proto:234]: or we simply use the concept binlog and make field as a optional parameter.

**Comment by xiaofan-luan** [comment]
Others all lgtm, please rebase and unify the concetpts of FieldBinLog and BinLog

File: internal/proto/data_coord.proto (hunk 1 above)
Comment (congqixia) [reply internal/proto/data_coord.proto:234]: > Just wondering, is it better to also FieldBinlog for delta logs and make Binlog a private concept inside FieldBinLog?

It's doable but `FieldID` will be redundant as each Binlog for same column will all have the same Field ID value.

The reason why DeltaLogs does not need field ID is that the delete entry is always applied on PK field.

It true that it looks ugly here. I will unify the `deltalogs` like other binlog fields.

**Comment by congqixia** [comment]
@xiaofan-luan Rebase and fixed. Could you please take a look again?

**Pull request merged**